### PR TITLE
Improve the docs for ordered options [ci skip]

### PR DESCRIPTION
**Pull request by hahmed**
### Summary

Improve the docs around ordered options to make it clearer.

### CODE REVIEW

File: activesupport/lib/active_support/ordered_options.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@
 require "active_support/core_ext/object/blank"
 
 module ActiveSupport
-  # Usually key value pairs are handled something like this:
+  # +OrderedOptions+ extends the +Hash+ object to make it easier to create or access keys.
+  # Usually we create key value pairs like:
```
Comment (jonathanhefner) [inline activesupport/lib/active_support/ordered_options.rb:7]: "Easier" might be debatable here.  What if we were more explicit?

```suggestion
  # +OrderedOptions+ inherits from +Hash+ and provides dynamic accessor methods.
  #
  # With a +Hash+, key-value pairs are typically managed like this:
```

File: activesupport/lib/active_support/ordered_options.rb [hunk 2]
```diff
@@ -12,7 +13,7 @@ module ActiveSupport
   #   h[:girl] # => 'Mary'
   #   h[:dog]  # => nil
   #
-  # Using +OrderedOptions+, the above code could be reduced to:
+  # Using +OrderedOptions+, the above is simplified so we can create and access keys easier:
```
Comment (jonathanhefner) [inline activesupport/lib/active_support/ordered_options.rb:16]:
```suggestion
  # Using +OrderedOptions+, the above code can be written as:
```

**Comment by hahmed** [comment]
In a number of places the method_missing is usually private or nodoc.

Should we also nodoc that here or is this intended?

https://github.com/rails/rails/blob/add1d7506865321eb5940e601098eab1a618f964/activesupport/lib/active_support/ordered_options.rb#L43

**Comment by jonathanhefner** [comment]
Hmm, I can only find [one nodoc `method_missing`](https://github.com/rails/rails/blob/e82c10b49ad9cb949f77297e6a814fd8f09d3c8f/activerecord/lib/active_record/migration.rb#L640).  Did you find more?  I do see a few that are documented, though ([example](https://github.com/rails/rails/blob/fbe2433be6e052a1acac63c7faf287c52ed3c5ba/activemodel/lib/active_model/attribute_methods.rb#L417-L427), [example](https://github.com/rails/rails/blob/fbe2433be6e052a1acac63c7faf287c52ed3c5ba/activesupport/lib/active_support/multibyte/chars.rb#L59-L60)).

**Comment by hahmed** [comment]
I think this one too -- https://github.com/rails/rails/blob/e82c10b49ad9cb949f77297e6a814fd8f09d3c8f/actionmailer/lib/action_mailer/base.rb#L585 

These are private methods -- https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/http/mime_type.rb#L308

Thanks for clearing that up, will squash the commits. 👍

**Comment by jonathanhefner** [comment]
Thank you @hahmed! :smile:

**Pull request merged**